app/uk/gov/hmrc/cgtpropertydisposals/models/ListUtils.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.models
+
+object ListUtils {
+
+  implicit class ListOps[A](private val l: List[A]) extends AnyVal {
+    def partitionWith[B, C](f: A => Either[B, C]): (List[B], List[C]) =
+      l.foldLeft(List.empty[B] -> List.empty[C]) {
+        case (acc, curr) =>
+          f(curr).fold(
+            b => (b :: acc._1) -> acc._2,
+            c => acc._1        -> (c :: acc._2)
+          )
+      }
+  }
+
```

Comment:
what's wrong with the standard lib partition function? :joy:

Reply:
the std library partition gives you two lists with the same type as the original list. I wanted to be able to partition into two lists with different types, so I went with this thing here.

Reply:
So this is just to check that valid asset types are coming from DES?

Reply:
it's used somewhere when transforming des returns to our complete returns in the transformer service as well